### DISCUSSION TITLE
show code run time in output

### DISCUSSION THREAD

**Discussion by Liripo** [quarto-dev/quarto-cli#5179] [Feature Requests]
![image](https://user-images.githubusercontent.com/54770415/231821714-5488a08f-8ac7-46f9-b766-b750c71d68ed.png)

As shown in the vscode IDE, if you can include the execution time of the code in the output, it will be very useful for writing test reports, or writing tutorial documents.

**Comment by mcanouil** [comment]
This can be easily achieved at least to measure computation, _i.e._, you can't really measure the whole rendering process because you would need to write the output then inject the run time in it[^1].

For example with `knitr` as the engine[^2]:

````qmd
---
title: "Quarto Demo"
format: html
---

```{r}
start_time <- Sys.time()
```

Something is wrong with the code below. It should be printing the current time, but it's not. Can you fix it?

```{r}
Sys.sleep(10)
```

```{r}
Sys.time() - start_time
```
````

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/231845398-534b70b4-41ec-438c-bc7c-efd7a61027d9.png">

[^1]: Actually, you can using pre/post-render script and a "place holder" in the output document (at least for format that can be edited) that you replace at the end with the time difference.
[^2]: you can make it more fancy

**Reply by Liripo** [reply]
Yes. I known this post: https://bookdown.org/yihui/rmarkdown-cookbook/time-chunk.html. But I think it would be interesting to natively support a kind of fancy output, such as including the running time in the output code block or below.

Of course I'll try your first suggestion myself.
